rust/azure_iot_operations_services/tests/azure_device_registry_client_tests.rs

```diff
@@ -0,0 +1,1161 @@
+                                    "Received unexpected additional asset observation notification"
+                                );
+                            }
+                        } else {
+                            log::info!("[{log_identifier}] Receiving no more notifications.");
+                            break;
+                        }
+                    }
+
+                    log::info!(
+                        "[{log_identifier}] Asset update notification receiver closed with count: {count}"
+                    );
+                    count
+                }
+            });
+
+            let response = azure_device_registry_client
+                .get_asset(
+                    DEVICE3.to_string(),
+                    ENDPOINT3.to_string(),
+                    ASSET_NAME3.to_string(),
+                    TIMEOUT,
+                )
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Get asset to update the status: {response:?}",);
+
+            patch_asset_specification(ASSET_NAME3, &update_desc);
+
+            // Wait for first notification with timeout (e.g., 30 seconds)
+            let notification_timeout = Duration::from_secs(30);
+            let notified_future = first_notification_notify.notified();
+            let did_receive_1_notification_or_timeout =
+                tokio::time::timeout(notification_timeout, notified_future).await;
+
+            azure_device_registry_client
+                .unobserve_asset_update_notifications(
+                    DEVICE3.to_string(),
+                    ENDPOINT3.to_string(),
+                    ASSET_NAME3.to_string(),
+                    TIMEOUT,
+                )
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Asset update unobservation: {:?}", ());
+
+            if did_receive_1_notification_or_timeout.is_ok() {
+                log::info!("[{log_identifier}] First notification received successfully");
+            }
+
+            patch_asset_specification(
+                ASSET_NAME3,
+                format!(
+                    "Patch specification update to NOT trigger notification after unobserve {}",
+                    Uuid::new_v4()
+                )
+                .as_str(),
+            );
+
+            let notification_count = match receive_notifications_task.await {
+                Ok(count) => {
+                    log::info!(
+                        "[{log_identifier}] Notification receiver task completed with count: {count}"
+                    );
+                    count
+                }
+                Err(e) => {
+                    log::error!(
+                        "[{log_identifier}] Notification receiver task failed on receiving unexpected counts: {e:?}"
+                    );
+                    panic!("Notification receiver task failed on receiving unexpected counts");
```

Comment:
this could also panic because the update doesn't match what we expected to be sent, so the message might need updating if you keep with having that method panic